src/index.js

```diff
@@ -458,8 +473,14 @@ class Libp2p extends EventEmitter {
 
     const announceFilter = this._options.addresses.announceFilter || ((multiaddrs) => multiaddrs)
 
+    // dedupe multiaddrs
```

Comment:
Per https://github.com/libp2p/js-libp2p/pull/810/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json#r534270687 we need to understand if the logic above is what we really want. Currently, (per 0.30) if `announceAddrs` were configured, only them will be advertised in the network. This will mean that the observed addresses will not be included. I think they should probably be included with the announceAddrs otherwise we would not even run the NAT Mgr if we had announceAddrs. Moreover, signed peer records will be created but their multiaddrs will be the same.
What do you think? @achingbrain @jacobheun

Reply:
The logic is changed - previously announceAddrs overrode allAddrs, subject to any announceFilter the user had configured.  Now announceAddrs overrides allAddrs, but is also joined with observedAddrs, and is subject to any announceFilter the user  has configured.